app/controllers/spam_reports_controller.rb

```diff
@@ -2,8 +2,8 @@ class SpamReportsController < ApplicationController
   wrap_parameters false
 
   def create
-    delivery_attempt = DeliveryAttempt.find(params[:reference])
-    SpamReportService.call(delivery_attempt)
+    email = DeliveryAttempt.find(params[:reference]).email
```

Comment:
Do we get a parameter of the email address? since we only hold on to DeliveryAttempts for a week this might 404 a lot. However if we used the email address we could remove the subscriptions even once we've deleted all the delivery attempts.